app/src/main/java/de/langerhans/odintools/tools/SettingsRepo.kt

```diff
@@ -0,0 +1,110 @@
+package de.langerhans.odintools.tools
+
+import de.langerhans.odintools.BuildConfig
+import javax.inject.Inject
+
+class SettingsRepo @Inject constructor(
+    private val executor: ShellExecutor,
+) {
+
+    init {
+        println("Settings init")
+        enableA11yService()
+        grantAllAppsPermission()
+        // Don't add to whitelist on debug builds, otherwise even Android Studio can't kill the app
+        if (!BuildConfig.DEBUG) {
+            addOdinToolsToWhitelist()
+        }
+    }
+
+    private fun enableA11yService() {
+        val currentServices =
+            executor.executeAsRoot("settings get secure enabled_accessibility_services")
+                .map { it ?: "" }
+                .getOrDefault("")
+
+        if (currentServices.contains(PACKAGE)) return
+
+        executor.executeAsRoot(
+            "settings put secure enabled_accessibility_services $PACKAGE/$PACKAGE.service.ForegroundAppWatcherService:$currentServices"
+                .trimEnd(':'),
+        )
+    }
+
+    private fun grantAllAppsPermission() {
+        executor.executeAsRoot("pm grant $PACKAGE android.permission.QUERY_ALL_PACKAGES")
+    }
+
+    private fun addOdinToolsToWhitelist() {
+        val currentWhitelist = whitelist
+        if (currentWhitelist.contains(PACKAGE)) return
+        val newWhitelist = "$PACKAGE,$currentWhitelist".trimEnd(',')
+        whitelist = newWhitelist
+    }
+
+    val buildVersion: String
+        get() = executor.getStringProperty(KEY_BUILD_VERSION, "")
+
+    val vendorName: String
+        get() = executor.getStringProperty(KEY_VENDOR_NAME, "")
+
+    var saturation: Float
+        get() = throw UnsupportedOperationException()
```

Comment:
I'm a bit torn on this. Style wise a property that can't even be read shouldn't be a property at all. This would only fail at runtime, not at compile time so I'd rather use a method for this setting tbh.

Reply:
Sounds good. Just to clarify, the current saturation value will be recovered from the user-defined setting (shared preferences).

Reply:
Yup that works. I also just looked into SurfaceFlinger and it reads `persist.sys.sf.color_saturation` on boot to apply the setting. So in theory we could write that and get rid off the persistence in OdinTools. Then again I like the ide of it not applying anymore when someone decides to uninstall OdinTools. Leaves the device in a clean state. 🤔